addons/mysql/templates/_helpers.tpl

```diff
@@ -123,4 +123,295 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 
 {{- define "mysql.imagePullPolicy" -}}
 {{ default "IfNotPresent" .Values.image.pullPolicy }}
+{{- end }}
+
+{{- define "mysql.spec.common" -}}
+provider: kubeblocks
+description: mysql component definition for Kubernetes
+serviceKind: mysql
+serviceVersion: 8.0.33
+updateStrategy: BestEffortParallel
+
+services:
+  - name: mysql-server
+    serviceName: mysql-server
+    spec:
+      ports:
+        - name: mysql
+          port: 3306
+          targetPort: mysql
+  - name: mysql
+    serviceName: mysql
+    podService: true
+    spec:
+      ports:
+        - name: mysql
+          port: 3306
+          targetPort: mysql
+
+scripts:
+  - name: mysql-scripts
+    templateRef: mysql-scripts
+    namespace: {{ .Release.Namespace }}
+    volumeName: scripts
+    defaultMode: 0555
+volumes:
+  - name: data
+    needSnapshot: true
+systemAccounts:
+  - name: root
+    initAccount: true
+    passwordGenerationPolicy:
+      length: 10
+      numDigits: 5
+      numSymbols: 0
+      letterCase: MixedCases
+vars:
+  - name: MYSQL_ROOT_USER
+    valueFrom:
+      credentialVarRef:
+        ## reference the current component definition name
+        compDef: {{ include "mysql.componentDefName" . }}
+        name: root
+        username: Required
+
+
+  - name: MYSQL_ROOT_PASSWORD
+    valueFrom:
+      credentialVarRef:
+        ## reference the current component definition name
+        compDef: {{ include "mysql.componentDefName" . }}
+        name: root
+        password: Required
+{{- end }}
+
+{{- define "mysql.spec.runtime.common" -}}
+- command:
+    - cp
+    - -r
+    - /bin/syncer
+    - /config
+    - /kubeblocks/
+  image: infracreate-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/syncer:latest
+  imagePullPolicy: {{ include "mysql.imagePullPolicy" . }}
+  name: init-syncer
+  volumeMounts:
+    - mountPath: /kubeblocks
+      name: kubeblocks
+- command:
+    - cp
+    - -r
+    - /xtrabackup-2.4
+    - /kubeblocks/xtrabackup
+  image: infracreate-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/syncer:mysql
+  imagePullPolicy: {{ default .Values.image.pullPolicy "IfNotPresent" }}
+  name: init-xtrabackup
+  volumeMounts:
+    - mountPath: /kubeblocks
+      name: kubeblocks
+{{- end }}
+
+{{- define "mysql-orc.spec.common"}}
+labels:
+  kubeblocks.io/ready-without-primary: "true"
+provider: kubeblocks
+description: mysql 5.7 component definition for Kubernetes
```

Comment:
dito. check service version, cofigs, scripts, and other.  as this template is used for both cmpd 5.7 and cmpd 8.0